fastplotlib/widgets/image.py

```diff
@@ -184,6 +197,56 @@ def current_index(self) -> Dict[str, int]:
         """
         return self._current_index
 
+    @property
+    def n_img_dims(self) -> list[int]:
+        """
+        list indicating the number of dimensions that contain image/single frame data for each data array.
+        if 2: data are grayscale, i.e. [x, y] dims, if 3: data are [x, y, c] where c is RGB or RGBA,
+        this is the complement of `n_scrollable_dims`
+        """
+        return self._n_img_dims
+
+    def _get_n_scrollable_dims(self, curr_arr: np.ndarray, rgb: bool) -> list[int]:
+        """
+        For a given ``array`` displayed in the ImageWidget, this function infers how many of the dimensions are
+        supported by sliders (aka scrollable). Ex: "xy" data has 0 scrollable dims, "txy" has 1, "tzxy" has 2.
+
+        Parameters
+        ----------
+        curr_arr: np.ndarray
+            np.ndarray or a list of array-like
+
+        rgb: bool
+            True if we view this as RGB(A) and False if grayscale
+
+        Returns
+        -------
+        list[int]
```

Comment:
```suggestion
        int
```
@apasarkar I'm editing this in-place, let me know if it's correct.